src/coap_io.c

```diff
@@ -734,9 +744,11 @@ struct in_pktinfo {
 
 #if defined(_WIN32)
 #include <mswsock.h>
-#if !defined(__MINGW32__)
+#if defined(__MINGW32__)
+static __thread LPFN_WSARECVMSG lpWSARecvMsg = NULL;
+#else
 static __declspec(thread) LPFN_WSARECVMSG lpWSARecvMsg = NULL;
-#endif /* ! __MINGW32__ */
+#endif /* defined(__MINGW32__) */
```

Comment:
This should then be reverted, so indicating the end of the code block to be what it is finishing.